server/models/Expense.js

```diff
@@ -141,6 +142,32 @@ export default function (Sequelize, DataTypes) {
       deletedAt: {
         type: DataTypes.DATE,
       },
+
+      tags: {
+        type: DataTypes.ARRAY(DataTypes.STRING),
+        set(tags) {
+          if (tags) {
+            tags = tags
+              .map(tag => {
+                if (tag) {
+                  const upperCase = tag.toUpperCase();
+                  const cleanTag = upperCase.trim().replace(/\s+/g, ' ');
+                  return stripHTML(cleanTag);
+                }
+              })
+              .filter(tag => {
+                return tag && tag.length > 0;
+              });
+          }
+
+          if (!tags || tags.length === 0) {
+            this.setDataValue('tags', null);
+          } else if (tags) {
+            this.setDataValue('tags', Array.from(new Set(tags)));
```

Comment:
Is that a de-duplication? In this case, why is it not done before? And why not use lodash?

Reply:
Doing that in the setter seems like the good place to me, that way you make sure that tags are always de-duplicated, no matter where & how you're creating/updating them. Agree that lodash could simplify the code.

Reply:
Doing that in the setter seems like the good place to me, that way you make sure that tags are always de-duplicated, no matter where & how you're creating/updating them. Agree that lodash could simplify the code.

Reply:
By earlier, I just wanted to say before `(!tags || tags.length === 0)` and not directly in the `setDataValue`.

Reply:
Good point!